shared_client.go

```diff
@@ -229,8 +378,14 @@ func handler(wg *sync.WaitGroup, client *Client, conn *Conn, requests chan reque
 		}
 	}()
 
+	deadline := time.NewTimer(0)
 	for {
+		// update timer
+		deadline.Reset(waiters.GetTimeout())
```

Comment:
`Reset` is not safe to use when the timer is not drained https://pkg.go.dev/time#Timer.Reset

How do we know that the time is drained here?

Reply:
Thanks for noting this, will have to work around this...